### ISSUE TITLE
ShapeView circle not rendering on iOS Devices (working in emulator)

### ISSUE THREAD

**Issue by BenReierson** [vincentgury/XFShapeView#4]
**Edit - I've actually just confirmed I'm able to see the circle render on physical iOS devices when connected in debug mode, just not when I've installed the adhoc build. 

I'm using a shapeview to draw a simple circle in a Xamarin Forms application. It's defined in Xaml and bound to a property in my VM using a value converter (enum -> color). This works fine on android (devices and emulators), and fine in the iOS emulators, but I see nothing on actual iOS devices. 

```
        <shapes:ShapeView x:Name="rarityCircle"
                          HorizontalOptions="End" VerticalOptions="Start" 
                          ShapeType="Circle" 
                          BorderColor="White" BorderWidth="2" 
                          Opacity=".7" 
                          Margin="5" 
                          HeightRequest="20" WidthRequest="20"
                          Color="{Binding Rarity, Converter={StaticResource Key=RarityColor}}}"/>
```

![image](https://cloud.githubusercontent.com/assets/6970511/19676900/863cbf7c-9ae2-11e6-869d-1aebca57c1ff.png)

Circles missing on iOS hardware:
![image](https://cloud.githubusercontent.com/assets/6970511/19676984/f9c847f4-9ae2-11e6-87f2-573da40a44f8.png)

**Comment by vincentgury** [comment]
Unfortunately, it will be hard to help you. As I don't have a paid Apple developer account, I can't deploy to physical device in Release mode.

I don't understand your second screenshot, can you make it bigger and give a little explanation?

Is the sample project working?

**Comment by BenReierson** [comment]
Yeah, I recognize it could be tricky to debug, but I really would like to
use this control. Do you have any confirmations it's been deployed and used
successfully in production on ios?

The picture is just meant to show that I have a listview of items and one
of the visual elements in a shapeview circle that changes color for each
item depending on a data bound property (as shown in the code I included).

I can maybe try and do an ad-hoc build of the sample tonight.

**Comment by vincentgury** [comment]
No I don't have such a confirmation. The library is quite new so I didn't get feedbacks.

As it's working in debug on ios, I suppose you added the Renderer.Init() in the AppDelegate. Do you confirm?

**Comment by BenReierson** [comment]
I'm definitely calling the Init function. I've temporarily switched to using boxviews in my ios project, but would like to return to using this control if it remains part of my visual overhaul.

**Comment by BenReierson** [comment]
Yes, I have the call to Init.

**Comment by vincentgury** [comment]
I tried to reproduce your issue without success in debug mode. Did you try the sample? Did you try to change the linker behavior?

Can you tell more about your development environment (device, ios version, ...)?

**Comment by BenReierson** [comment]
It continues to work in debug mode for me too. I don't have enough
experience with building for iOS to know exactly what differs with release
mode. I haven't tried changing the link behavior, but I can try that later
today. Currently, debug mode compiles with 'don't link' but release mode is
set to 'link framework sdks only'. I just did a quick test of running in
debug with 'link framework sdks only'. The shapeview still worked, so there
might be something else different happening in release mode.

**Comment by BenReierson** [comment]
It continues to work in debug mode for me too. I don't have enough
experience with building for iOS to know exactly what differs with release
mode. I haven't tried changing the link behavior, but I can try that later
today. Currently, debug mode compiles with 'don't link' but release mode is
set to 'link framework sdks only'. I just did a quick test of running in
debug with 'link framework sdks only' and then 'link all'. The shapeview
still worked in both cases, so there might be something else different
happening in release mode.

**Comment by vincentgury** [comment]
Thanks for the testing.

For now, I can't help you more but I soon as I take a paid apple dev subscription, I will dig deeper to fix it.

Feel free to fork the project and have a look at the source code if you need a quicker fix.

**Comment by rostislavrotaru** [comment]
Putting Init() function in Renderer class will not help. You must delete the Renderers class and put the Init() function in ShapeRenderer class, and add the reference to FShapeView.dll into Nuget package for iOS. It will work on Release mode. I've tested and it works.

In iOS project (AppDelegate.cs) call ShapeRenderer.Init()

**Comment by rostislavrotaru** [comment]
I am using your package in a release app and with this workaround it works, but I did it manually. Waiting for a new NuGet release

**Comment by BenReierson** [comment]
Sounds promising. @rostislavrotaru thanks for figuring this out. I'm sure a pull request with the change would be appreciated. 

@vincentgury can you confirm you'll be able to update the project. Let me know if you want to me to test a prerelease of the nuget, or a branch of the code. Thanks.

**Comment by vincentgury** [comment]
@rostislavrotaru thanks a lot for your investigation and detailed explanation!

@BenReierson sure I will update the project. I will do my best to upload a prerelease nuget package this week. I will need your help for the testing.

**Comment by vincentgury** [comment]
I just released a new UNTESTED version (1.0.5-pre) of the package with the fix.

Can you please try it and tell me if the fix works?

**Comment by BenReierson** [comment]
Hi Vincent,
I have to admit, I've moved on from using this control in my current
project, but I will try to pull an older build to test this.

**Comment by henningms** [comment]
Can't get 1.0.5-pre to work on device either. Works in debug though.

**Comment by henningms** [comment]
I created a PR with the necessary (albeit small) change to make it work on iOS release mode

**Comment by vincentgury** [comment]
Thanks a lot. I accepted the PR but I still can't test. I will publish a new nuget package including your fix later today cos I am not at home right now.

When done, do you have some time to test it?

**Comment by vincentgury** [comment]
The new package is uploaded: VG.XFShapeView 1.0.5-pre2

Hope this time it will work knowing I had to build in release mode for simulator (no paid dev account still)

**Comment by djordjedjukic** [comment]
I released test flight with 1.0.5-pre2 and it's working now. Great job!
